File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/gui/providers/examples/bossbar/BossbarExample.java
package com.gmail.borlandlp.minigamesdtools.arena.gui.providers.examples.bossbar;

import com.gmail.borlandlp.minigamesdtools.arena.ArenaBase;
import com.gmail.borlandlp.minigamesdtools.arena.gui.providers.GUIProvider;
import com.gmail.borlandlp.minigamesdtools.arena.team.TeamProvider;
import org.bukkit.Bukkit;
import org.bukkit.boss.BarColor;
import org.bukkit.boss.BarStyle;
import org.bukkit.boss.BossBar;
import org.bukkit.entity.Player;

import java.util.Random;

public class BossbarExample extends GUIProvider {
    private String message = null;
    private float percentage = 0.0F;
    private BossBar bossBar;

    public BossbarExample(ArenaBase arena) {
        super(arena);
    }

    public BossbarExample() {}

    public String getMessage() {
        return this.message;
    }

    public void setPercentage(float percent) {
        this.percentage = percent;
    }

    public float getPercentage() {
        return this.percentage;
    }

    @Override
    public void gameEnded() {
        if(this.bossBar == null) {
            return;
        }

        for(TeamProvider team : this.getArena().getTeamController().getTeams()) {
            for (Player player : team.getPlayers()) {
                if(player != null) this.bossBar.removePlayer(player);
            }
        }
    }

    @Override
    public void onInit() {

    }

    @Override
    public void beforeGameStarting() {
        BarColor color = BarColor.values()[ (new Random()).nextInt(BarColor.values().length - 1) ];
        BarStyle style = BarStyle.values()[ (new Random()).nextInt(BarStyle.values().length - 1) ];
        this.bossBar = Bukkit.createBossBar("BossBarExample", color, style);
        this.bossBar.setProgress((new Random()).nextDouble());
    }

    public void update() {
        this.bossBar.setColor(BarColor.values()[ (new Random()).nextInt(BarColor.values().length - 1) ]);
        this.bossBar.setStyle(BarStyle.values()[ (new Random()).nextInt(BarStyle.values().length - 1) ]);

        for(TeamProvider team : this.getArena().getTeamController().getTeams()) {
            for (Player player : team.getPlayers()) {
                if (player != null && !this.bossBar.getPlayers().contains(player)) {
                    bossBar.addPlayer(player);
                    bossBar.setVisible(true);
                }
            }
        }
    }

    @Override
    public void beforeRoundStarting() {

    }

    @Override
    public void onRoundEnd() {

    }

    public void setMessage(String message) {
        this.message = message;
    }

    public void removeMessage() {
        this.message = null;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/gui/providers/examples/bossbar/BossbarExample.kt
package com.gmail.borlandlp.minigamesdtools.arena.gui.providers.examples.bossbar

import com.gmail.borlandlp.minigamesdtools.arena.ArenaBase
import com.gmail.borlandlp.minigamesdtools.arena.gui.providers.GUIProvider
import org.bukkit.Bukkit
import org.bukkit.boss.BarColor
import org.bukkit.boss.BarStyle
import org.bukkit.boss.BossBar
import java.util.*

class BossbarExample : GUIProvider {
    var message: String? = null
    var percentage = 0.0f
    private var bossBar: BossBar? = null

    constructor(arena: ArenaBase?) : super(arena) {}
    constructor() {}

    override fun gameEnded() {
        if (bossBar == null) {
            return
        }
        for (team in arena!!.teamController!!.teams) {
            for (player in team.players) {
                bossBar!!.removePlayer(player)
            }
        }
    }

    override fun onInit() {}
    override fun beforeGameStarting() {
        val color = BarColor.values()[Random().nextInt(BarColor.values().size - 1)]
        val style = BarStyle.values()[Random().nextInt(BarStyle.values().size - 1)]
        bossBar = Bukkit.createBossBar("BossBarExample", color, style)
        bossBar!!.progress = Random().nextDouble()
    }

    override fun update() {
        bossBar!!.color = BarColor.values()[Random().nextInt(BarColor.values().size - 1)]
        bossBar!!.style = BarStyle.values()[Random().nextInt(BarStyle.values().size - 1)]
        for (team in arena!!.teamController!!.teams) {
            for (player in team.players) {
                if (!bossBar!!.players.contains(player)) {
                    bossBar!!.addPlayer(player)
                    bossBar!!.isVisible = true
                }
            }
        }
    }

    override fun beforeRoundStarting() {}
    override fun onRoundEnd() {}

    fun removeMessage() {
        message = null
    }
}